File: updated_code/code/java/121-best-time-to-buy-and-sell-stock.java
package code;

public class Solution {
    /**
     * You are given an array prices where prices[i] is the price of a given stock on the ith day.
     * 
     * You want to maximize your profit by choosing a **single day** to buy one stock and choosing a **different day in the future** to sell that stock.
     * 
     * Return _the maximum profit you can achieve from this transaction_. If you cannot achieve any profit, return `0`.
     * 
     * @param prices an array of stock prices
     * @return the maximum profit that can be achieved from a single transaction
     */
    public int maxProfit(int[] prices) {
        int minPrice = prices[0];
        int maxProfit = 0;

        for(int i = 1; i < prices.length; i++) {
            // Avoid buying at a price that is higher than what we already have
            if(prices[i] < minPrice) {
                minPrice = prices[i];
            } else if(prices[i] - minPrice > maxProfit) {
                maxProfit = prices[i] - minPrice;
            }
        }

        return maxProfit;
    }
}

File: updated_code/code/python/121-best-time-to-buy-and-sell-stock.py
Python
from typing import List

class Solution:
    """
    A class that solves the problem of finding the maximum profit from buying and selling a stock once.

    Attributes:
        max_profit (int): The maximum profit that can be achieved from buying and selling a stock once.

    Methods:
        maxProfit(prices: List[int]) -> int: The maximum profit that can be achieved from buying and selling a stock once.
    """

    def __init__(self):
        """
        Initializes the solution.
        """

    def maxProfit(self, prices: List[int]) -> int:
        """
        Finds the maximum profit that can be achieved from buying and selling a stock once.

        Args:
            prices (List[int]): A list of prices of a stock on different days.

        Returns:
            int: The maximum profit that can be achieved from buying and selling a stock once.
        """
        if not prices:
            return 0

        min_price = prices[0]
        max_profit = 0

        for price in prices:
            if price < min_price:
                min_price = price
            elif price - min_price > max_profit:
                max_profit = price - min_price

        return max_profit